src/Microsoft.Diagnostics.Monitoring.EventPipe/EventPipeStreamProvider.cs

```diff
@@ -28,7 +30,43 @@ public async Task<Stream> ProcessEvents(DiagnosticsClient client, TimeSpan durat
             EventPipeSession session = null;
             try
             {
-                session = await client.StartEventPipeSessionAsync(_sourceConfig.GetProviders(), _sourceConfig.RequestRundown, _sourceConfig.BufferSizeInMB, cancellationToken).ConfigureAwait(false);
+                IEnumerable<EventPipeProvider> providers = _sourceConfig.GetProviders();
+                int bufferSizeInMB = _sourceConfig.BufferSizeInMB;
+                long rundownKeyword = _sourceConfig.RundownKeyword;
+                RetryStrategy retryStrategy = _sourceConfig.RetryStrategy;
+                bool retry = true;
+                while (retry)
```

Comment:
Consider unrolling this since there are only really two attempts to avoid the loop and the extra state:

```cs
try
{
    EventPipeSessionConfiguration config = new(_sourceConfig.GetProviders(), _sourceConfig.BufferSizeInMB, (_sourceConfig.RundownKeyword != 0), true, _sourceConfig.RundownKeyword);
    session = await client.StartEventPipeSessionAsync(config, cancellationToken).ConfigureAwait(false);
}
catch (UnsupportedCommandException e) when (...)
{
}

EventPipeSessionConfiguration config = new(_sourceConfig.GetProviders(), _sourceConfig.BufferSizeInMB);
session = await client.StartEventPipeSessionAsync(config, cancellationToken).ConfigureAwait(false);

```